core/sql/generator/GenPreCode.cpp

```diff
@@ -10715,7 +10808,10 @@ RelExpr * PhysicalFastExtract::preCodeGen (Generator * generator,
   if (nodeIsPreCodeGenned())
     return this;
 
-  generator->setIsFastExtract(TRUE);  
+  if (getIsMainQueryOperator())
+    generator->setIsFastExtract(TRUE);
+  else
+    generator->setContainsFastExtract(TRUE);
```

Comment:
I would have thought that "contains" would include the "is" case... but perhaps mathematically speaking we are talking about proper containment?